.github/workflows/tics.yaml

```diff
@@ -14,32 +14,46 @@ jobs:
         with:
           fetch-depth: 0
 
-      - name: Install dependencies
+      - name: Install coverage tools
         run: |
-          sudo apt update
-          python -m pip install --upgrade pip
-          # pin tox to the current major version to avoid
-          # workflows breaking all at once when a new major version is released.
-          python -m pip install 'tox<5'
-
-        # The snap is built already in the check workflow and available as an artifact,
-        # but we want this TICS workflow to be independent,
-        # so we can run the workflow periodically or on demand.
-      - name: Build the snap
-        id: build
-        uses: canonical/action-build@v1
-
-      - name: Test with tox and produce coverage report
-        # Create an environment in /tmp for the TICS action to use.
-        run: tox --workdir /tmp/tox -e tics
+          pip install coverage
+
+      - name: Download unit test coverage artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: coverage-unit
+          path: ./artifacts
+        continue-on-error: true  # Don’t fail if artifact is missing
+
+      - name: Download functional test coverage artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: coverage-functional
+          path: ./artifacts
```

Comment:
have you try `pattern: coverage-*` + `merge-multiple: true`, this might combine the result without having manually download two artifact. https://github.com/actions/download-artifact?tab=readme-ov-file#inputs

Reply:
Something important that Ray raised before: we'll get coverage files from every run - eg. when we have tests on different architectures.

We can ensure we don't upload conflicting files by adding some unique identifier to each coverage file, and using this kind of wildcard download.

Reply:
Good point. I've added the arch in the name of the artifact.
I've also applied Ray's suggestion, so now all artifact are downloaded in a single step